README.adoc

```diff
@@ -1,97 +1,114 @@
-      -->
-      <version>0.0.1-SRC-revision-66ea95d890531f4eaaa5aa04a9b1c69b409dcd0b</version>
-    </dependency>
-  </dependencies>
-
-  <build>
-    <plugins>
-     [...]
-
-      <plugin>
-        <groupId>org.l2x6.maven.srcdeps</groupId>
-        <artifactId>srcdeps-maven-plugin</artifactId>
-        <version>0.0.1</version>
-        <extensions>true</extensions>
-        <configuration>
-          <repositories>
-            <repository>
-              <!-- a build tree unique id chosen by you -->
-              <id>org.hawkular.cmdgw</id>
-              <selectors>
-                <!-- groupId of the dependencies that come from this repo
-                     The syntax may change in the future,
-                     see https://github.com/l2x6/srcdeps-maven-plugin/issues/10 -->
-                <selector>org.hawkular.cmdgw</selector>
-              </selectors>
-              <!-- The url that is passed to checkout mojo of SCM Plugin,
-                   see https://maven.apache.org/scm/maven-scm-plugin/checkout-mojo.html -->
-              <url>scm:git:https://github.com/hawkular/hawkular-command-gateway.git</url>
-            </repository>
-          </repositories>
-        </configuration>
-      </plugin>
-
-    </plugins>
-  </build>
-
-  [...]
+ Various settings have to be configured in `.mvn/srcdeps.yaml` file. Especially, the mapping from groupId's of
+ dependencies to SCM repositiries where their sources reside.
 
+[source,yaml]
 ----
+configModelVersion: 1.0
+forwardProperties:
+- myForwardProperty
+builderIo:
+  stdin: inherit
+  stdout: inherit
+  stderr: inherit
+skip: false
+skipTests: false
+sourcesDirectory: ~/.m2/srcdeps
+verbosity: info
+repositories:
+  hawkularCommons:
+    selectors:
+    - org.hawkular.commons # if one of the selectors matches the groupId of a -SRC- dependency,
+                           # then the urls and buildArguments are used to build that dependency
+    urls:
+    - git:https://github.com/hawkular/hawkular-commons.git
+    buildArguments:
+    - -Dcheckstyle.skip=true
+    - -Dlicense.skip=true
+    - -Drevapi.skip=true
+----
+
+
+=== Legacy note
+
+Originally, the present project provided a Maven plugin called `srcdeps-maven-plugin` that worked fairly well in most situations, but it turned out soon that the Maven Plugin API is not rich enough to allow for catching dependencies of
+all possible kinds. Most notably, the given project's parent pom and imports in `dependencyManagement` sections were
+resolved by Maven even before the plugin was initialized. Thus, if the given project's parent pom had a `-SRC-`
+version, the build failed with a missing artifact error message, simply because it happened before
+`srcdeps-maven-plugin` could do anything.
+
+An outdated presentation about the abandoned `srcdeps-maven-plugin`: http://ppalaga.github.io/presentations/150929-srcdeps-maven-plugin/150929-srcdeps-maven-plugin.html
+
 
-== Why to use Srcdeps Plugin
+== Why to use Srcdeps tools
```

Comment:
Great piece of text, let me put it there! Added a new commit: https://github.com/ppalaga/srcdeps/commit/d09cac5775e39e99c23f7b97bf6497d52a057fc1